for_pr_review.ts

```diff
@@ -0,0 +1,244 @@
+      host: this.configService.get<string>('REDIS_HOST'),
+      port: this.configService.get<number>('REDIS_PORT'),
+      password: this.configService.get<string>('REDIS_PASSWORD'),
+    }
+  
+  );
+
+    // this.maxClients = maxClients;
+    // this.clients = [];
+  }
+
+ getClient(): Redis {
+    return this.clientA;
+  }
+
+getClientForLock(): Redis {
+  return this.clientB
+}
+
+redlock(client): Redlock {
+  const redlock = new Redlock(
+    [client],
+    {
+      //시간 동기화
+      driftFactor: 0.01,
+      //재시도 횟수
+      retryCount: 10,
+      //재시도 간 간격
+      retryDelay: 200,
+      //최대 딜레이 시간
+      retryJitter:300,
+      //잠금 만료 연장 하는 시점
+      automaticExtensionThreshold: 100,
+    }
+  )
+  return redlock
+}
+
+}
+
+##src/payments/payments.service.ts - pay
+import { InjectRepository } from '@nestjs/typeorm';
+import { DataSource, Repository } from 'typeorm';
+import { Payments } from './entities/payments.entity';
+import { BadRequestException, Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { Point } from '../point/entities/point.entity';
+import { Users } from '../user/entities/user.entitiy';
+import { Orders } from '../orders/entities/orders.entity';
+import { CreatePaymentDto } from '../orders/dto/create-payment.dto';
+import { validation } from '../common/pipe/validationPipe';
+import { OrdersDetails } from '../orders/entities/ordersdetails.entity';
+import { Stocks } from '../goods/entities/stocks.entity';
+import logger from '../common/log/logger';
+import { Goods } from '../goods/entities/goods.entity';
+import { Status } from '../orders/types/order.type';
+import _ from 'lodash';
+import { RedisService } from '../redis/redis.service';
+import Redlock from 'redlock';
+
+@Injectable()
+export class PaymentsService {
+  constructor(
+    @InjectRepository(Payments)
+    private paymentsRepository: Repository<Payments>,
+    @InjectRepository(Point)
+    private pointRepository: Repository<Point>,
+    @InjectRepository(Users)
+    private usersRepository: Repository<Users>,
+    @InjectRepository(Orders)
+    private ordersRepository: Repository<Orders>,
+    @InjectRepository(OrdersDetails)
+    private ordersDetailsRepository: Repository<OrdersDetails>,
+    private readonly dataSource: DataSource,
+    private readonly redisService: RedisService
+  ) { }
+
+  // 전체적으로 추가적인 보안요소 필요
+  async pay(
+    userId: number,
+    createPaymentDto: CreatePaymentDto // 포스트맨의 body,
+  ) {
+    if (_.isNil(userId) || userId == 0) {
+      const error = new BadRequestException('잘못된 요청입니다!') 
+      logger.errorLogger(error, `userId = ${userId}, createPaymentDto = ${createPaymentDto}`)
+      throw error
+    }
+    await validation(CreatePaymentDto, createPaymentDto)
+    const { orders_id } = createPaymentDto
+    const queryRunner = this.dataSource.createQueryRunner();
+
+    const order = await queryRunner.manager.findOne(Orders, {
+      where: {
+        id: orders_id, user_id: userId
```

Comment:
```suggestion
        id: orders_id, userId,
```

entity 가 snake_case 를 사용하는 것 같은데, js 에서는 camelCase 가 기본입니다!

Reply:
넵 다음에는 꼭 camelCase로 하도록 하겠습니다. 저 근데 rdbms에서 컬럼에 대문자를 인식시킬 수 있나여??

Reply:
column 속성중에 name 을 사용하면 됩니당

Reply:
감사합니다!